### PR TITLE
Reduce number of feeds to restore from for Razor/Blazor assets

### PR DESCRIPTION
**Pull request by captainsafia**
For Blazor and Razor test assets, we only look for packages in the `dotnet6` and the official NuGet feed.

### CODE REVIEW

**Review by mmitche — APPROVED** [review]

File: src/Tests/Microsoft.NET.TestFramework/AspNetSdkTest.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+namespace Microsoft.NET.TestFramework
+{
+    public abstract class AspNetSdkTest : SdkTest
+    {
+        private readonly IEnumerable<System.Reflection.AssemblyMetadataAttribute> TestAssemblyMetadata;
+        public readonly string DefaultTfm;
+
+        private List<string> NuGetFeedsRetained = new List<string>(){"dotnet-public", "dotnet6", "dotnet6-transport"};
+
+        protected AspNetSdkTest(ITestOutputHelper log) : base(log) {
```
Comment (pranavkm) [inline src/Tests/Microsoft.NET.TestFramework/AspNetSdkTest.cs:24]: Formatting

File: src/Tests/Microsoft.NET.TestFramework/AspNetSdkTest.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+using NuGet.Configuration;
+
+namespace Microsoft.NET.TestFramework
+{
+    public abstract class AspNetSdkTest : SdkTest
+    {
+        private readonly IEnumerable<System.Reflection.AssemblyMetadataAttribute> TestAssemblyMetadata;
+        public readonly string DefaultTfm;
+
+        private List<string> NuGetFeedsRetained = new List<string>(){"dotnet-public", "dotnet6", "dotnet6-transport"};
```
Comment (pranavkm) [inline src/Tests/Microsoft.NET.TestFramework/AspNetSdkTest.cs:22]: I think you might want to say any feed that starts with `dotnet6`.

File: src/Tests/Microsoft.NET.TestFramework/AspNetSdkTest.cs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+using System.Xml.Linq;
+using Microsoft.NET.TestFramework;
+using Xunit.Abstractions;
+using NuGet.Configuration;
+
+namespace Microsoft.NET.TestFramework
+{
+    public abstract class AspNetSdkTest : SdkTest
+    {
+        private readonly IEnumerable<System.Reflection.AssemblyMetadataAttribute> TestAssemblyMetadata;
```
Comment (pranavkm) [inline src/Tests/Microsoft.NET.TestFramework/AspNetSdkTest.cs:19]: Unusual casing choice for fields

File: src/Tests/Microsoft.NET.TestFramework/AspNetSdkTest.cs [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+            var assembly = Assembly.GetCallingAssembly();
+            TestAssemblyMetadata = assembly.GetCustomAttributes<AssemblyMetadataAttribute>();
+            DefaultTfm = TestAssemblyMetadata.SingleOrDefault(a => a.Key == "AspNetTestTfm").Value;
+        }
+
+        private void CopyAndModifyNugetConfig(string projectDirectory)
+        {
+            // Copy the config at the root of the repo to `projectDirectory`
+            var repoRoot = TestAssemblyMetadata.SingleOrDefault(a => a.Key == "ArtifactsTmpDir").Value;
+            var configAtRoot = Path.Combine(repoRoot, "nuget.config");
```
Comment (pranavkm) [inline src/Tests/Microsoft.NET.TestFramework/AspNetSdkTest.cs:34]: Does this fail on case-sensitive oses? The file in the root has a very specific case: https://github.com/dotnet/sdk/blob/master/NuGet.config (Btw, we could fix that up)

**Review by pranavkm — APPROVED** [review]

**Pull request closed without merging**